### PR TITLE
Draw and eat dots

### PR DESCRIPTION
**Pull request by MirkoFer-622**

### CODE REVIEW

File: src/main/java/pacman/DrawComponents.java [hunk 1]
```diff
@@ -29,18 +31,30 @@ public DrawComponents() {
...
-        for (int i = 0; i < walls.size(); i++) {
-        g.setColor(Color.BLUE);
-        g.drawRect(walls.get(i).getX(), walls.get(i).getY(), 30, 30);
+        for (int i=0;i<walls.size(); i++) {
+            g.setColor(Color.BLUE);
+            g.fillRect(walls.get(i).getX(), walls.get(i).getY(), 30, 30);  
+        }
+    }
+    public void drawDots(Graphics g) {
+        for (int i=0;i<dots.size(); i++) {
```
Comment (maryangela) [inline src/main/java/pacman/DrawComponents.java]: add spaces

File: src/main/java/pacman/DrawComponents.java [hunk 2]
```diff
@@ -29,18 +31,30 @@ public DrawComponents() {
...
+            g.setColor(Color.WHITE);
+            if((dots.get(i).isSpecialFood()==true)&&(dots.get(i).exist==true)){
```
Comment (maryangela) [inline src/main/java/pacman/DrawComponents.java]: add spaces

File: src/main/java/pacman/Pacman.java [hunk 3]
```diff
@@ -49,6 +53,15 @@ public boolean existDistintPacman(int xx, int yy) {
         }
         return exist;
     }
+    //Verify if exist a dot in that position
+    public boolean existDot(int xx, int yy) {
+        boolean exist=false;
+        for(Position i: dots){
+            if(i.getX() ==xx && i.getY()==yy)
```
Comment (maryangela) [inline src/main/java/pacman/Pacman.java]: add spaces

File: src/main/java/pacman/Pacman.java [hunk 4]
```diff
@@ -53,6 +53,15 @@ public boolean existDistintPacman(int xx, int yy) {
         }
         return exist;
     }
+    //Verify if exist a dot in that position
```
Comment (Laura4lilavati) [inline src/main/java/pacman/Pacman.java]: Please change the comment to: Check if there is a dot in that position

File: src/main/java/pacman/Pacman.java [hunk 5]
```diff
@@ -53,6 +53,15 @@ public boolean existDistintPacman(int xx, int yy) {
...
+    public boolean existDot(int xx, int yy) {
+        boolean exist=false;
+        for(Position i: dots){
+            if(i.getX() ==xx && i.getY()==yy)
+                exist= true;
```
Comment (Laura4lilavati) [inline src/main/java/pacman/Pacman.java]: Please add a single space before the equal

File: src/main/java/pacman/Dot.java [hunk 6]
```diff
@@ -1,12 +1,23 @@
...
         this.isSpecial = false;
     }
-
-    private boolean isSpecialFood() {
+    
+    public boolean isSpecialFood(){
         return this.isSpecial;
     }
+
+    public boolean setNotSpecial(){
```
Comment (J0c4) [inline src/main/java/pacman/Dot.java]: This method could be void

File: src/main/java/pacman/Dot.java [hunk 7]
```diff
@@ -1,12 +1,23 @@
...
+        this.isSpecial=false;
+        return true;
+    }
+
+    public boolean setSpecial(){
```
Comment (J0c4) [inline src/main/java/pacman/Dot.java]: Same here

File: src/main/java/pacman/DrawComponents.java [hunk 8]
```diff
@@ -29,18 +31,30 @@ public DrawComponents() {
...
-        g.setColor(Color.BLUE);
-        g.drawRect(walls.get(i).getX(), walls.get(i).getY(), 30, 30);
+            g.setColor(Color.BLUE);
+            g.fillRect(walls.get(i).getX(), walls.get(i).getY(), 30, 30);  
+        }
+    }
+    public void drawDots(Graphics g) {
+        for (int i = 0; i < dots.size(); i++) {
+            g.setColor(Color.WHITE);
+            if((dots.get(i).isSpecialFood() == true) && (dots.get(i).exist == true)) {
```
Comment (J0c4) [inline src/main/java/pacman/DrawComponents.java]: you don't `== true`

File: src/main/java/pacman/DrawComponents.java [hunk 9]
```diff
@@ -55,32 +69,72 @@ public void drawGhosts(Graphics g) {
...
     }
-
-    //Method to move pacman to right 10 pixels
+  
+    //Method to move pacman to right 10 pixelsAnd Check if there is a dot or superdot
     public void moveRight() {
         x = pacman.right();
         y = y;
+        for (Dot i: dots) {
+        if((pacman.existDot(x,y))&&(i.getX() == x && i.getY() == y)) {
```
Comment (J0c4) [inline src/main/java/pacman/DrawComponents.java]: Please add 4 spaces for the if

File: src/main/java/pacman/DrawComponents.java [hunk 10]
```diff
@@ -55,32 +69,72 @@ public void drawGhosts(Graphics g) {
...
         repaint();
     }
-
-    //Method to move pacman to right 10 pixels
+  
+    //Method to move pacman to right 10 pixelsAnd Check if there is a dot or superdot
     public void moveRight() {
         x = pacman.right();
         y = y;
+        for (Dot i: dots) {
```
Comment (J0c4) [inline src/main/java/pacman/DrawComponents.java]: instead of `i` you can use `dot`

File: src/main/java/pacman/DrawComponents.java [hunk 11]
```diff
@@ -55,32 +72,72 @@ public void drawGhosts(Graphics g) {
...
-    //Method to move pacman to up 10 pixels
+    //Method to move pacman to up 10 pixels And Check if there is a dot or superdot
     public void moveUp() {
         y = pacman.up();
         x = x;
+        for (Dot i: dots) {
+            if(pacman.existDot(x,y)&&(i.getX() == x && i.getY() == y)) {
+                  i.exist=false;
+            }
+            if (((pacman.existDot(x,y))&&(i.getX() == x && i.getY() == y))&&(i.isSpecial)) {
```
Comment (roht0515) [inline src/main/java/pacman/DrawComponents.java]: please add spaces around "&&"

File: src/main/java/pacman/DrawComponents.java [hunk 12]
```diff
@@ -55,32 +72,72 @@ public void drawGhosts(Graphics g) {
...
-    //Method to move pacman to down 10 pixels
+    //Method to move pacman to down 10 pixels And Check if there is a dot or superdot
     public void moveDown() {
         y = pacman.down();
         x = x;
+        for (Dot i: dots) {
+            if(pacman.existDot(x,y)&&(i.getX() == x && i.getY() == y)) {
+                  i.exist=false;
+            }
+            if (((pacman.existDot(x,y))&&(i.getX() == x && i.getY() == y))&&(i.isSpecial)){
```
Comment (roht0515) [inline src/main/java/pacman/DrawComponents.java]: please, add spaces around "&&"

File: src/main/java/pacman/DrawComponents.java [hunk 13]
```diff
@@ -55,32 +72,72 @@ public void drawGhosts(Graphics g) {
...
         x = pacman.left();
         y = y;
+        for (Dot i: dots) {
+            if(pacman.existDot(x,y)&&(i.getX() == x && i.getY() == y)) {
+                  i.exist=false;
+            }
+            if (((pacman.existDot(x,y))&&(i.getX() == x && i.getY() == y))&&(i.isSpecial)){
+                i.exist=false;
+                ghost1.changeEatable();
+                ghost2.changeEatable();
```
Comment (roht0515) [inline src/main/java/pacman/DrawComponents.java]: please add spaces aroud "&&" 
and here 
i.exist=false;
i.exist = false;

File: src/main/java/pacman/DrawComponents.java [hunk 14]
```diff
@@ -55,32 +72,72 @@ public void drawGhosts(Graphics g) {
...
+      }
         repaint();
     }
 
-    //Method to move pacman to left 10 pixels
+    //Method to move pacman to left 10 pixels And Check if there is a dot or superdot
     public void moveLeft() {
         x = pacman.left();
         y = y;
+        for (Dot i: dots) {
```
Comment (roht0515) [inline src/main/java/pacman/DrawComponents.java]: change the name of the variable "i", could be dot

File: src/main/java/pacman/DrawComponents.java (hunk 2 above)
Comment (julia-jpg) [reply src/main/java/pacman/DrawComponents.java]: No need to put "== true" after dots.get(i).exist  
and 
you should not access the attributes of another class, only its methods "dots.get(i).exist "

File: src/main/java/pacman/DrawComponents.java [hunk 15]
```diff
@@ -29,18 +31,33 @@ public DrawComponents() {
...
         for (int i = 0; i < walls.size(); i++) {
-        g.setColor(Color.BLUE);
-        g.drawRect(walls.get(i).getX(), walls.get(i).getY(), 30, 30);
+            g.setColor(Color.BLUE);
+            g.fillRect(walls.get(i).getX(), walls.get(i).getY(), 30, 30);  
+        }
+    }
+
+    public void drawDots(Graphics g) {
+        for (int i = 0; i < dots.size(); i++) {
```
Comment (J0c4) [inline src/main/java/pacman/DrawComponents.java]: You can use a foreach here

File: src/main/java/pacman/DrawComponents.java [hunk 16]
```diff
@@ -55,32 +72,72 @@ public void drawGhosts(Graphics g) {
...
+    //Method to move pacman to down 10 pixels And Check if there is a dot or superdot
     public void moveDown() {
         y = pacman.down();
         x = x;
+        for (Dot dot : dots) {
+            if(pacman.existDot(x,y) && (dot.getX() == x && dot.getY() == y)) {
+                dot.exist = false;
+            }
+            if (((pacman.existDot(x,y)) && (dot.getX() == x && dot.getY() == y)) && (dot.isSpecial)){
+                dot.exist = false;
```
Comment (J0c4) [inline src/main/java/pacman/DrawComponents.java]: please encapsulate the behavior of Dot to stop existing

**Pull request merged**